source/Octopus.Tentacle/Commands/ImportCertificateCommand.cs

```diff
@@ -28,17 +29,18 @@ public ImportCertificateCommand(Lazy<IWritableTentacleConfiguration> tentacleCon
 
             Options.Add("r|from-registry", "Import the Octopus Tentacle 1.x certificate from the Windows registry", v => fromRegistry = true);
             Options.Add("f|from-file=", "Import a certificate from the specified file generated by the new-certificate command or a Personal Information Exchange (PFX) file", v => importFile = v);
+            Options.Add("b|from-base64=", "Import a certificate from a base64 formatted string", v => importBase64 = v);
             Options.Add("pw|pfx-password=", "Personal Information Exchange (PFX) private key password", v => importPfxPassword = v, sensitive: true);
         }
 
         protected override void Start()
         {
             base.Start();
-            if (!fromRegistry && string.IsNullOrWhiteSpace(importFile))
+            if (!fromRegistry && string.IsNullOrWhiteSpace(importFile) && string.IsNullOrWhiteSpace(importBase64))
                 throw new ControlledFailureException("Please specify the certificate to import.");
 
-            if (fromRegistry && !string.IsNullOrWhiteSpace(importFile))
-                throw new ControlledFailureException("Please specify only one of either from-registry or from-file.");
+            if (fromRegistry && (!string.IsNullOrWhiteSpace(importFile) || !string.IsNullOrWhiteSpace(importBase64)))
```

Comment:
It looks like the purpose of this guard is to only allow 1 of the 3 options, no more.

However, if `fromRegistry` is false, I think this `if` statement will happily allow `importFile` and `importBase64` to both be set.

Reply:
Could we write a test for this as well?

Reply:
Ah yup I think you are right. Ill fix it up